### DISCUSSION TITLE
Welcome to dfkreport Discussions!

### DISCUSSION THREAD

**Discussion by pwillworth** [pwillworth/dfkreport#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas for improvements.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by kazamaj9999** [comment]
Hi there, thanks for the website!  Your report is missing the Gen 0 sale and associated Jewel sales etc on 9/30/2021 for some reason.  It comes up on https://kingdoms.tax/ website but not yours.

**Reply by pwillworth** [reply]
Thanks for the heads up, I'll look up the info for that so it's detected as soon as I can

**Reply by pwillworth** [reply]
Gen 0 hero sales should now be included in newly generated reports!

**Reply by kazamaj9999** [reply]
Thanks! I tried running new report with new dates and it still didn't show up.  Just FYI (I just manually entered).  Thanks again

**Reply by pwillworth** [reply]
Is it specifically a gen 0 hero purchase where you bought a hero crystal that you are expecting to see, or some other gen0 airdrop/jewel sale activity?

**Reply by kazamaj9999** [reply]
it was the initial gen 0 purchase that happened on 9/30 for gen 0 hero crystal

**Reply by pwillworth** [reply]
Ok, I think I found a reason why a few of them might not have got populated and resolved that.  Please try generating again with new dates and let me know.

**Reply by kazamaj9999** [reply]
Still didn't populate

**Reply by pwillworth** [reply]
Ah dang, they should all be there now.  Can you DM me the hero ID or wallet address so I can investigate further?  Need more info to be able to dig deeper.

**Comment by kazamaj9999** [comment]
Also, second question, anyway to combine transactions from two different wallets?  Some people transferred accounts to add a hardware wallet etc.  Thanks!

**Reply by pwillworth** [reply]
Currently the best you could do is to run both and export and combine them.  That is a feature that I had on my list to add soon though!

**Reply by pwillworth** [reply]
@kazamaj9999 been a while, but multi-wallet support has just been added on Lila's Ledger

**Comment by BigMonkeyCryptoGit** [comment]
Just a big well done and thank you. This is awesome.

**Comment by zepedie** [comment]
Hi, congrats on the bounty win and thanks for putting this tool together.

It looks like the three decimal point is too short for WBTC so many transactions show as $0 value.

**Reply by pwillworth** [reply]
Hey there, thanks - and thank you for the youtube content I enjoy it -
Yeah, I struggled a bit with the decimals on the report display.  I ultimately ended with rounding things on the display but keeping the complete and exact numbers in the csv export.  Perhaps I should implement something more variable for the report display with increased decimals for high value tokens like BTC and ETH.

**Reply by zepedie** [reply]
Thanks for the video feedback :-)
Hmmm, so looking at the csv, I wonder if it's a different issue. I think it's just an issue with my sold BTC transactions; the cost and gain showing as x.xxE-07 (or effectively zero). But, if I sold something else for BTC then I think it's okay. Attaching a couple of screenshots.

![Screen Shot 2022-01-21 at 11 12 43 am](https://user-images.githubusercontent.com/98074539/150447759-bef3acdc-0ed2-405e-8184-69001905c45e.png)
![Screen Shot 2022-01-21 at 11 15 28 am](https://user-images.githubusercontent.com/98074539/150448004-2d45c04f-1420-4b3e-9347-8627d836bdb9.png)

**Reply by pwillworth** [reply]
So, looks like there was a bug and I was not converting the BTC decimals correctly!  Missed that in testing, thanks for bringing it up.  I have corrected it now, but you would need to generate a new report by picking a different date range to see the fix reflected or it will just download your prior generated report with the bad data immediately.

**Comment by lordgiammus** [comment]
Hi, first of all thank you for your hard work and congrats on the bounty! I think this tool will be really useful.
The report is missing a hero summon that I did on January 23rd (3 days ago). The summon appears on https://kingdoms.tax/ and kingdom.watch also reports the correct number of heroes in my wallet. All the other transactions seem to be up to date. Do you know what's causing the problem?
Moreover, is there a way to calculate the total value of DFK assets in USD at a certain date (December 31st, 2021 for example)? I mean, the sum of the total USD value of jewels, LPs, Heroes, Inventory, etc. That would really be useful.

**Reply by pwillworth** [reply]
Hi! Glad to hear from you.  I have seen some issues with misses in the real time tracking of summons/sales that I have been resolving with another process that comes and does another pass to catch anything that was missed.  That can lag behind a few days though.  I'm actively working to make that less duct-tape ish, but there is a catch up running now, so I think that missing one would show up soon.

My approach for this app was to satisfy the bounty which calls specifically for evaluating historical activity, so I did not design in any kind of features for assessing the current total worth at any point in time.  There are already apps that do a good job of that (kingdom.watch), so it feels a little redundant to put in that functionality.

**Reply by lordgiammus** [reply]
Ok, thank you for your reply! I'll check in a few days and I'll let you know if the hero summon is still missing.

**Comment by lalala-lea** [comment]
Hello! Thank you for creating this report  - it's so helpful. In looking at the Tax Report I noticed that some of the transactions seem to be using the incorrect cost basis. As you can see in the screenshot below, all of the ONE that was being sold was purchased on the same day. The first couple transactions look correct/reasonable for the cost basis, but the following ones don't. 
For example the third one down is suggesting that the cost of the ONE being sold is ~$0.63 and the next one almost $1 and the one after >$1. Could you explain what might be happening and is there anyway to fix the issue?

Editing to add - there are other transactions that have the same issue with the cost basis - this is just a sample of some of them.

![Screen Shot 2022-03-11 at 9 04 20 AM](https://user-images.githubusercontent.com/101420419/157916444-497d5856-1b5b-4271-b59c-438627124005.jpg)

**Reply by pwillworth** [reply]
Hi, thanks for spotting this.  I believe I have found a bug that was adding costs wrong in some edge cases where cost basis is split among many transactions in a certain way.  Please try generating your report again now, and verify this is corrected.

**Reply by lalala-lea** [reply]
Thank you so much for the quick reply! I generated the report again and it now looks correct to me. This report must be so complicated to build- thank you so much for doing it and addressing issues as they come up. Really appreciate it!

**Comment by bluidyPCish** [comment]
Hello! 

A lot of us have been having issues with using the exported CSV files with any of the crypto tax services. Namely koinly, tokentax etc. Could you let us know if there's some specific way the file needs to be sent or such? Danke.

**Reply by pwillworth** [reply]
Hi!  First thing I would check is, make sure you are exporting from the the Transactions tab and selecting Koinly Universal for CSV Format.  The export from the "Tax Report" tab is more of a processed output than tax services are expecting.  Please make sure you are doing those two things.  If you are, and are still having issues, I'll need to know more specifically what the issues are to know the cause or resolution.

**Comment by bluidyPCish** [comment]
Thank you for the response back! 

Will do this! 

Stay Thirsty Always…

**Comment by Aka-Chuck** [comment]
Hey there, thank you so much for creating this site/service. It's going to be such a huge help for filing taxes! I really appreciate it and I'm sure a lot of others do too.

I'm just dropping in to report a possible bug. I noticed some strange behavior for stamina potion related transactions - some trades seem to be calculated as $0 or even just really weird large numbers. I'll attach a couple of examples - if it's something I'm misunderstanding or wrong about sorry in advance.

![Example 1](https://user-images.githubusercontent.com/101685485/158507971-204ec1ae-97d5-428e-87fb-89a2624576d2.png)

![Example 2](https://user-images.githubusercontent.com/101685485/158507997-d1347d2b-5d57-4f13-8a59-f14242104919.png)

**Reply by pwillworth** [reply]
Hi, thanks, I'm glad it's going to be helpful for you!

Yes - unfortunately for the most part, you wont see accurate prices for items like potions and eggs for now.  The source I am using to lookup prices does not have data for that kind of stuff.  I would like to do an enhancement to look in another place to get those kind of prices, so keep an eye open for that update some time in the future.

**Comment by ShockLabCodebase** [comment]
Hey,

Love the site, especially the new Perilous Journey features. I have noticed that several of my perished heroes aren't showing up though, and some that are showing don't have a $0 cost basis. Let me know if you need any more details. Happy to DM you my 0x if that helps.

Cheers.

**Reply by pwillworth** [reply]
Hi, glad you're liking it!  Yes, I have seen this too occasionally.  It seems that sometimes when Harmony is being testy, some transactions are left out when we go fetch the list of them for your account from harmony api.  When this happens, typically you can run the report again (with a slightly changed date range to force it to fetch the list again) and it will fill in those gaps.  Let me know if that doesn't work for you.

**Comment by hysticali** [comment]
Hey,  Just want to thank you for continuing to update this. It is hugely appreciated. For your donation address, can we send you in game items like eggs or do you prefer we sell them and send you Jewel or Crystal?

**Reply by pwillworth** [reply]
Hi, thanks, It's awesome to hear that people are still finding it useful as I try to keep it up to date with all the changes coming out.

Anything including items is appreciated, though ultimately I have to convert/sell items into tokens so I can put it towards paying the server hosting costs.  So, if you just send jewel/crystal it saves me that step.
